cg/services/sample_service/sample_service.py

```diff
@@ -0,0 +1,43 @@
+from cg.services.sample_service.utils import (
+    get_cancel_comment,
+    get_confirmation_message,
+)
+from cg.store.models import User
+from cg.store.store import Store
+
+
+class SampleService:
+
+    def __init__(self, store: Store):
+        self.store = store
+
+    def cancel_sample(self, sample_id: int, user_email: str) -> list[str]:
+        """Cancel a single sample and return a list of all cases it used to be in."""
+        case_ids: list[str] = self.store.get_case_ids_with_sample(sample_id)
```

Comment:
The returned list here is very confusing and to me it does not serve that much of a purpose. Could we not let this method only cancel the sample? Then we could just store the case ids before calling it.

Reply:
Resolved in https://github.com/Clinical-Genomics/cg/pull/3507/commits/41c8b7b29beffb69bd26234e2ab22bd87627fedb

Reply:
Yeah, I agree, for me this would be

1. `case_ids = [case.internal_id] for case in sample.links`
2. `cancel_sample(sample)`
3. `cancel_empty_cases(case_ids)`
4. `remaining_cases = get_remaining_cases(case_ids)`
5. `return [case.internal_id for case in remaining_cases]`

Reply:
Cleaned up 👍